src/Common/src/CoreLib/System/MemoryExtensions.cs

```diff
@@ -245,12 +251,34 @@ public static bool SequenceEqual<T>(this Span<T> first, ReadOnlySpan<T> second)
             where T : IEquatable<T>
         {
             int length = first.Length;
-            if (typeof(T) == typeof(byte))
+            if (typeof(T) == typeof(byte) || typeof(T) == typeof(sbyte))
                 return length == second.Length &&
                 SpanHelpers.SequenceEqual(
```

Comment:
Should this call SequenceEqualBytes directly?

Reply:
It could - I added the extra `SequenceEquals` override precisely so that we'd get the right behavior either way, though. There's a number of existing code paths that the `T==byte` check - I didn't want to add a `(nuint)` cast to all of them.

Reply:
It seems odd, though, that all the other types call SequenceEqualBytes except for when T==byte.

Reply:
The byte case is the only one that doesn't have to scale the length by the size of the type, and hence, include a vacuous cast to `nuint` as part of the call. That's why it makes sense to let that case have the more succinct and natural version of the callsite.